### PR TITLE
disable some scores not used for MS1 transitions; set max iterations …

### PR DESCRIPTION
**Pull request by timosachsenberg**
…for faster EMG fits

## Description

Do not merge yet as test is failing. Processing time was reduced from 6 minutes to 4 on a larger dataset.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by timosachsenberg** [comment]
Currently, most time is spent in the quadratic FeatureFinderAlgorithmMetaboIdent::findOverlappingFeatures_. 
An n log n implementation could certainly be possible.

**Comment by timosachsenberg** [comment]
This determining the set of 2D-interval intersection problem has many solutions.
Recursive Dimensional Clustering from Game Programming Gems 2 comes to my mind that could be used to solve this quickly and is still easily implemented.

**Comment by jpfeuffer** [comment]
I thought we already have a KDTree implementation. Just use it.

**Comment by jpfeuffer** [comment]
We are probably even doing the exact same thing somewhere else.
Refactor and reuse.

**Comment by timosachsenberg** [comment]
kd-tree would work but might require quite a lot of extra memory. If I find time, I can check - the code from MapAlignerKDTree should basically give us what we want if we give every feature a different map index.

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmMetaboIdent.cpp [hunk 1]
```diff
@@ -182,9 +182,29 @@ namespace OpenMS
...
+    // around the apex, to complete isotopic envelopes (and therefore make this score more robust).
+
+    if ((elution_model_ != "none") || (!candidates_out_.empty()))
+    {
+      params.setValue("Scores:use_elution_model_score", "false"); // TODO: test if this works for requantificiation
+      params.setValue("write_convex_hull", "true");
+    }
+    else // no elution model
+    {
+      params.setValue("write_convex_hull", "false");
```
Comment (timosachsenberg) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmMetaboIdent.cpp:205]: this currently fails with a missing convex hull error @jpfeuffer do you recall that you had to deal with that in plfq?

**Comment by timosachsenberg** [comment]
time for extraction and quantificaiton of 106394 features went down fom 6:30 min to 3:14 mins

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmMetaboIdent.cpp (hunk 1 above)
Comment (jpfeuffer) [reply src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmMetaboIdent.cpp]: I never disabled completely.

**Comment by timosachsenberg** [comment]
tests pass... ready for review

**Comment by jpfeuffer** [comment]
What about FeatureFinderIdent?

**Comment by jpfeuffer** [comment]
I think it should benefit from moving, too

**Comment by timosachsenberg** [comment]
It doesn't have this kind of overlap resolution. @hendrikweisser is that right?

**Comment by hendrikweisser** [comment]
>> What about FeatureFinderIdent?

> It doesn't have this kind of overlap resolution. @hendrikweisser is that right?

Yes, correct.

**Comment by jpfeuffer** [comment]
I see. Out of curiosity, why is it more/only important in Metabo?

**Comment by hendrikweisser** [comment]
> why is it more/only important in Metabo?

It came up as a key issue in the use case for which FFMetId was developed (quantification of modified RNA nucleosides). It's a combination of how common overlaps are and how much you care about resulting quantification errors.

If we now have an efficient way of detecting overlaps, it may be worthwhile porting it over to FFId, at least as an option. It would be especially useful if we could come up with a method to still quantify overlapping features somewhat robustly. (Instead of the current approaches of ignoring overlaps altogether - in FFId - or removing overlapping features - in FFMetId).

**Comment by timosachsenberg** [comment]
Note: It is not efficient in terms of complexity. Still n*n. But at least not as heavy on the memory reallocations which gave a huge speed up.

**Comment by timosachsenberg** [comment]
with the upcoming fix by Julianus I think it is good to merge this now as the PR seems kind of complete  on its own

**Pull request merged**